### ISSUE TITLE
[sqlite3] Build error on x64-windows-static-md

### ISSUE THREAD

**Issue by LucasXu0** [microsoft/vcpkg#41308]
Hi, Team. 

I'm encountering an failure in one of my Windows GitHub Actions workflows. The failure has started occurring recently and seems to be random. I'd appreciate any insights or guidance on how to resolve this issue.

```Log
C:\vcpkg\triplets\community\x64-windows-static-md.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
-- Downloading https://sqlite.org/2024/sqlite-autoconf-3460100.tar.gz -> sqlite-autoconf-3460100.zip...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from C:\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: C:\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
warning: Download failed -- retrying after 1000ms
warning: Download failed -- retrying after 2000ms
warning: Download failed -- retrying after 4000ms
error: Missing sqlite-autoconf-3460100.zip and downloads are blocked by x-block-origin.
error: https://sqlite.org/2024/sqlite-autoconf-3460100.tar.gz: WinHttpSendRequest failed with exit code 10106
error: https://sqlite.org/2024/sqlite-autoconf-3460100.tar.gz: WinHttpSendRequest failed with exit code 10106
error: https://sqlite.org/2024/sqlite-autoconf-3460100.tar.gz: WinHttpSendRequest failed with exit code 10106
error: https://sqlite.org/2024/sqlite-autoconf-3460100.tar.gz: WinHttpSendRequest failed with exit code 10106
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1030): 
[DEBUG] Time in subprocesses: 0us
[DEBUG] Time in parsing JSON: 2us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 832us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 7 s (7028615us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.
      
      3. Your proxy's remote server is out of service.
      
      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.
      

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  ports/sqlite3/portfile.cmake:4 (vcpkg_download_distfile)
  scripts/ports.cmake:192 (include)


error: building sqlite3:x64-windows-static-md failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle sqlite3:x64-windows-static-md: 7.1 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+sqlite3
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[sqlite3]+Build+error+on+x64-windows-static-md&body=Copy+issue+body+from+C%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
You can also submit an issue by running (GitHub CLI must be installed):
  gh issue create -R microsoft/vcpkg --title "[sqlite3] Build failure on x64-windows-static-md" --body-file C:/vcpkg/installed/vcpkg/issue_body.md
Error while executing command, exit code: 1
```

**Comment by dg0yt** [comment]
Probably https://github.com/microsoft/vcpkg/issues/41199#issuecomment-2378255699

Update and bootstrap vcpkg to resolve the issue.

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. Please reopen this issue if this is still a problem for you.
